File: app/src/main/java/edu/schoolapp/SettingsActivity.java
package edu.schoolapp;

import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.AsyncTask;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.graphics.drawable.RoundedBitmapDrawable;
import android.support.v4.graphics.drawable.RoundedBitmapDrawableFactory;
import android.support.v7.app.AppCompatActivity;
import android.view.MenuItem;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.google.android.gms.auth.api.Auth;
import com.google.android.gms.auth.api.signin.GoogleSignIn;
import com.google.android.gms.auth.api.signin.GoogleSignInApi;
import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
import com.google.android.gms.common.api.GoogleApiClient;
import com.google.android.gms.common.api.ResultCallback;
import com.google.android.gms.common.api.Status;

import java.net.URL;

public class SettingsActivity extends AppCompatActivity {

    private ImageView userImageView;
    private TextView userNameTextView;
    private TextView emailTextView;
    private TextView managePrimaryMenuTextView;
    private TextView logoutTextView;

    private GoogleApiClient googleApiClient;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);

        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        userImageView = findViewById(R.id.setting_user_image_view);
        userNameTextView = findViewById(R.id.settings_user_name_text_view);
        emailTextView = findViewById(R.id.settings_email_text_view);
        managePrimaryMenuTextView = findViewById(R.id.manage_text_view);
        logoutTextView = findViewById(R.id.logout_text_view);

        new DownloadImageAsync(userImageView).execute(SchoolApp.getProfileUrl());
        userNameTextView.setText(SchoolApp.getName());
        emailTextView.setText(SchoolApp.getEmail());

        managePrimaryMenuTextView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(SettingsActivity.this, ManageListFragment.class));
            }
        });

        logoutTextView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (googleApiClient.isConnected()) {
                    Auth.GoogleSignInApi.signOut(googleApiClient);
                    googleApiClient.disconnect();
                    googleApiClient.connect();

                    SchoolApp.clearSharedPreference();

                    Intent intent = new Intent(SettingsActivity.this, LoginActivity.class);
                    intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
                    startActivity(intent);
                }
            }
        });
    }

    @Override
    protected void onStart() {
        super.onStart();
        GoogleSignInOptions gso = new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
                .requestEmail()
                .build();
        googleApiClient = new GoogleApiClient.Builder(this)
                .addApi(Auth.GOOGLE_SIGN_IN_API, gso)
                .build();
        googleApiClient.connect();
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
        }
        return true;
    }


    //Inner Class
    class DownloadImageAsync extends AsyncTask<String, Void, Bitmap> {

        private ImageView userImage;

        DownloadImageAsync(ImageView userImage) {
            this.userImage = userImage;
        }

        @Override
        protected Bitmap doInBackground(String... urls) {
            Bitmap bitmap = null;
            try {
                bitmap = BitmapFactory.decodeStream(new URL(urls[0]).openStream());
            } catch (Exception e) {
                e.printStackTrace();
            }
            return bitmap;
        }

        protected void onPostExecute(Bitmap bitmap) {
            RoundedBitmapDrawable roundedBitmapDrawable = RoundedBitmapDrawableFactory.create(getResources(), bitmap);
            roundedBitmapDrawable.setCircular(true);
            userImage.setImageDrawable(roundedBitmapDrawable);
        }
    }
}

File: app/src/main/java/edu/schoolapp/SettingsActivity.kt
package edu.schoolapp

import android.content.Intent
import android.graphics.BitmapFactory
import android.os.Bundle
import android.view.MenuItem
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.core.graphics.drawable.RoundedBitmapDrawableFactory
import androidx.lifecycle.lifecycleScope
import edu.schoolapp.googleSingIn.GoogleAuthenticationService
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import java.net.URL

class SettingsActivity : AppCompatActivity() {
    private lateinit var userImageView: ImageView
    private lateinit var userNameTextView: TextView
    private lateinit var emailTextView: TextView
    private lateinit var managePrimaryMenuTextView: TextView
    private lateinit var logoutTextView: TextView

    private val googleAuthenticationService = GoogleAuthenticationService()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        userImageView = findViewById(R.id.setting_user_image_view)
        userNameTextView = findViewById(R.id.settings_user_name_text_view)
        emailTextView = findViewById(R.id.settings_email_text_view)
        managePrimaryMenuTextView = findViewById(R.id.manage_text_view)
        logoutTextView = findViewById(R.id.logout_text_view)

        userNameTextView.text = SchoolApp.name
        emailTextView.text = SchoolApp.email
        managePrimaryMenuTextView.setOnClickListener {
            startActivity(
                Intent(
                    this@SettingsActivity,
                    ManageListFragment::class.java
                )
            )
        }

        logoutTextView.setOnClickListener {
            googleAuthenticationService.logout()
            SchoolApp.clearSharedPreference()
            val intent = Intent(this@SettingsActivity, LoginActivity::class.java)
            intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP
            startActivity(intent)
        }

        downloadProfileImage()
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
        }
        return true
    }

    private fun downloadProfileImage() {
        lifecycleScope.launch(Dispatchers.IO) {
            val bitmap = BitmapFactory.decodeStream(URL(SchoolApp.profileUrl).openStream())
            val roundedBitmapDrawable = RoundedBitmapDrawableFactory.create(resources, bitmap)
            roundedBitmapDrawable.isCircular = true
            userImageView.setImageDrawable(roundedBitmapDrawable)
        }
    }
}